### PR TITLE
Delete `TxVotes` and `VotingEntry` and use `VotingProcedures` instead

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Delete `TxVotes` and `VotingEntry` and use `VotingProcedures` instead
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

`VotingProcedures` is able to replace both `VotingEntry` and `TxVotes`.

It is a `newtype` wrapper around the ledger types so creating generators for it is very easy.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/Governance/Actions/VotingEntry.hs
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Governance/Actions/VotingEntry.hs:1]: :tada:

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 1]
```diff
@@ -2718,8 +2719,8 @@ fromLedgerTxBody sbe scriptValidity body scriptdata mAux =
       , txMetadata
       , txAuxScripts
       , txScriptValidity    = scriptValidity
-      , txGovernanceActions = fromLedgerProposalProcedure sbe body
-      , txVotes             = fromLedgerTxVotes sbe body
+      , txGovernanceActions = fromLedgerProposalProcedure   sbe body
+      , txVotingProcedures             = fromLedgerVotingProcedures    sbe body
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/TxBody.hs:2723]: Some strange spacing here

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 2]
```diff
@@ -3430,7 +3427,7 @@ getByronTxBodyContent (Annotated Byron.UnsafeTx{txInputs, txOutputs} _) =
   , txMintValue         = TxMintNone
   , txScriptValidity    = TxScriptValidityNone
   , txGovernanceActions = TxGovernanceActionsNone
-  , txVotes             = TxVotesNone
+  , txVotingProcedures             = Nothing
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/TxBody.hs:3430]: Spacing

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 3]
```diff
@@ -2743,24 +2744,20 @@ fromLedgerProposalProcedure sbe body =
...
-        $ foldMap (\(voter, innerMap) ->
-              Map.mapKeys (\govActId -> (fromVoterRole (conwayEraOnwardsToShelleyBasedEra w) voter, GovernanceActionId govActId))
-            $ Map.map VotingProcedure innerMap
-          )
-        $ (Map.toList . Gov.unVotingProcedures)
-        $ body_ ^. L.votingProceduresTxBodyL
+        $ Just
+        $ Featured w
+        $ VotingProcedures
+        $ body ^. L.votingProceduresTxBodyL)
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/TxBody.hs:2759]: :tada:

**Review by Jimbo4350 — APPROVED** [review]
LGTM! Minor comments

**Pull request merged**